kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableFeatures.java

```diff
@@ -48,19 +45,19 @@ public class TableFeatures {
   ////////////////////
 
   public static void validateReadSupportedTable(
-      Protocol protocol, Metadata metadata, String tablePath) {
+      Protocol protocol, Optional<Metadata> metadata, String tablePath) {
```

Comment:
I think it's common convention for "optional" params to go at the end of the param list